// worker-timers/build/es5/bundle.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('worker-timers-broker')) :
    typeof define === 'function' && define.amd ? define(['exports', 'worker-timers-broker'], factory) :
    (global = global || self, factory(global.workerTimers = {}, global.workerTimersBroker));
}(this, (function (exports, workerTimersBroker) { 'use strict';

    var workerTimers = null;
    var createLoadWorkerTimers = function createLoadWorkerTimers(load, worker) {
      return function () {
        if (workerTimers !== null) {
          return workerTimers;
        }

        var blob = new Blob([worker], {
          type: 'application/javascript; charset=utf-8'
        });
        var url = URL.createObjectURL(blob);
        workerTimers = load(url); // Bug #1: Edge doesn't like the URL to be revoked directly.

        workerTimers.setTimeout(function () {
          return URL.revokeObjectURL(url);
        }, 0);
        return workerTimers;
      };
    };

    // This is the minified and stringified code of the worker-timers-worker package.
    var worker = "!function(e){var t={};function r(n){if(t[n])return t[n].exports;var o=t[n]={i:n,l:!1,exports:{}};return e[n].call(o.exports,o,o.exports,r),o.l=!0,o.exports}r.m=e,r.c=t,r.d=function(e,t,n){r.o(e,t)||Object.defineProperty(e,t,{enumerable:!0,get:n})},r.r=function(e){\"undefined\"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:\"Module\"}),Object.defineProperty(e,\"__esModule\",{value:!0})},r.t=function(e,t){if(1&t&&(e=r(e)),8&t)return e;if(4&t&&\"object\"==typeof e&&e&&e.__esModule)return e;var n=Object.create(null);if(r.r(n),Object.defineProperty(n,\"default\",{enumerable:!0,value:e}),2&t&&\"string\"!=typeof e)for(var o in e)r.d(n,o,function(t){return e[t]}.bind(null,o));return n},r.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return r.d(t,\"a\",t),t},r.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},r.p=\"\",r(r.s=0)}([function(e,t,r){\"use strict\";r.r(t);r(1)},function(e,t,r){var n,o;void 0===(o=\"function\"==typeof(n=function(){\"use strict\";var e=new Map,t=new Map,r=function(e,t){var r,n;if(\"performance\"in self){var o=performance.now();r=o,n=e-Math.max(0,o-t)}else r=Date.now(),n=e;return{expected:r+n,remainingDelay:n}},n=function e(t,r,n,o){var i=\"performance\"in self?performance.now():Date.now();i>n?postMessage({id:null,method:\"call\",params:{timerId:r,timerType:o}}):t.set(r,setTimeout(e,n-i,t,r,n,o))};addEventListener(\"message\",(function(o){var i=o.data;try{if(\"clear\"===i.method){var a=i.id,u=i.params,s=u.timerId,c=u.timerType;if(\"interval\"===c)!function(t){var r=e.get(t);if(void 0===r)throw new Error('There is no interval scheduled with the given id \"'.concat(t,'\".'));clearTimeout(r),e.delete(t)}(s),postMessage({error:null,id:a});else{if(\"timeout\"!==c)throw new Error('The given type \"'.concat(c,'\" is not supported'));!function(e){var r=t.get(e);if(void 0===r)throw new Error('There is no timeout scheduled with the given id \"'.concat(e,'\".'));clearTimeout(r),t.delete(e)}(s),postMessage({error:null,id:a})}}else{if(\"set\"!==i.method)throw new Error('The given method \"'.concat(i.method,'\" is not supported'));var l=i.params,f=l.delay,d=l.now,p=l.timerId,m=l.timerType;if(\"interval\"===m)!function(t,o,i){var a=r(t,i),u=a.expected,s=a.remainingDelay;e.set(o,setTimeout(n,s,e,o,u,\"interval\"))}(f,p,d);else{if(\"timeout\"!==m)throw new Error('The given type \"'.concat(m,'\" is not supported'));!function(e,o,i){var a=r(e,i),u=a.expected,s=a.remainingDelay;t.set(o,setTimeout(n,s,t,o,u,\"timeout\"))}(f,p,d)}}}catch(e){postMessage({error:{message:e.message},id:i.id,result:null})}}))})?n.call(t,r,t,e):n)||(e.exports=o)}]);"; // tslint:disable-line:max-line-length

    var loadWorkerTimers = createLoadWorkerTimers(workerTimersBroker.load, worker);
    var clearInterval = function clearInterval(timerId) {
      return loadWorkerTimers().clearInterval(timerId);
    };
    var clearTimeout = function clearTimeout(timerId) {
      return loadWorkerTimers().clearTimeout(timerId);
    };
    var setInterval = function setInterval(func, delay) {
      return loadWorkerTimers().setInterval(func, delay);
    };
    var setTimeout = function setTimeout(func, delay) {
      return loadWorkerTimers().setTimeout(func, delay);
    };

    exports.clearInterval = clearInterval;
    exports.clearTimeout = clearTimeout;
    exports.setInterval = setInterval;
    exports.setTimeout = setTimeout;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
